src/actions/userActions.js

```diff
@@ -82,6 +109,43 @@ export type VerificationPhoneAction = {
   oneTimePassword: string,
 }
 
+export const verifyEmailAction = (walletId: string, code: string) => {
+  return async (dispatch: Dispatch, getState: GetState, api: SDKWrapper) => {
+    dispatch(sendingOneTimePasswordAction());
+
+    const response = await api.verifyEmail({
+      walletId,
+      oneTimePassword: code,
+    });
+
+    const { responseStatus } = response;
+
+    if (responseStatus === 200) {
+      dispatch(logEventAction('email_verified'));
+
+      dispatch({ type: USER_EMAIL_VERIFIED });
+      dispatch({
+        type: ADD_NOTIFICATION,
+        payload: {
+          message: 'Email verification was successful',
+          title: 'Validation successful',
+          messageType: 'success',
+        },
+      });
+    } else {
+      dispatch(otpPasswordSentAction());
```

Comment:
Can you explain the logic here? Why we mark it as sent only when error occurs?

Reply:
This is so we go back to the state we had before calling `dispatch(sendingOneTimePasswordAction());`. The state chain, on failure scenario would be: `sent` -> `sending` -> `sent`.